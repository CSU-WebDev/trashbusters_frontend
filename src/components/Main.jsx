import React from 'react'
import Pitch from './Pitch'
import About from './About'
import TrashMap from './TrashMap'
import Contact from './Contact'
import Footer from './Footer'
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

export default function Main() {
    return (
        <div class="wrapper">
            <Container fluid class="center">
                <Row>
                    <Col sm={1}></Col>
                    <Col>
            <Pitch />
            <About />
            <TrashMap />
            <Contact />
            <Footer />
                    </Col>
                    <Col sm={1}></Col>
                </Row>
            </Container>
        </div>
    );
}
