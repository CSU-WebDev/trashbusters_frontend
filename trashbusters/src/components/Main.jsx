import React from 'react'
import Pitch from './Pitch'
import About from './About'
import TrashMap from './TrashMap'
import Contact from './Contact'
import Footer from './Footer'

export default function Main() {
    return (
        <>
            <Pitch />
            <About />
            <TrashMap />
            <Contact />
            <Footer />
        </>
    );
}