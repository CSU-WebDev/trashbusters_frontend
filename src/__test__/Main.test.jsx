import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Main from '../components/Main';

describe('Main component', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(<Main />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders child components', () => {
    render(<Main />);
    const mainContainer = screen.getByTestId('main');
    expect(mainContainer).toBeInTheDocument();

    expect(screen.getByTestId('pitch-container')).toBeInTheDocument();
    expect(screen.getByTestId('about-container')).toBeInTheDocument();
    expect(screen.getByTestId('trashmap-container')).toBeInTheDocument();
    expect(screen.getByTestId('contact-container')).toBeInTheDocument();
  });
});
