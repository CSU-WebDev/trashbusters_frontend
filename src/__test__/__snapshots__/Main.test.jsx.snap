// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container-fluid"
    data-testid="main"
    id="main"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="mt-5 box container-fluid"
          data-testid="pitch-container"
          id="pitch-container"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <h3
                class="text-center"
              >
                Which do you prefer???
              </h3>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col"
            >
              <div
                class="mt-4 mb-4 container"
                data-testid="pics-container"
                id="pics-container"
              >
                <div
                  class="justify-content-center row"
                  data-testid="pics-row"
                >
                  <div
                    class="align-items-center col-md-6"
                    data-testid="pics-col-1"
                    s="12"
                  >
                    <img
                      class="pic img-fluid"
                      data-testid="pic"
                      src="litter-before.jpg"
                    />
                  </div>
                  <div
                    class="align-items-center col-md-6"
                    data-testid="pics-col-2"
                    s="12"
                  >
                    <img
                      class="pic img-fluid"
                      data-testid="pic"
                      src="litter-after.jpg"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="box container-fluid"
          data-testid="about-container"
          id="about-container"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <h3
                class="text-center"
              >
                About
              </h3>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col"
            >
              <p
                data-testid="about-paragraph"
                id="about-paragraph"
              >
                Litter is 
                <i>
                  never
                </i>
                 fun to look at. Join the fight and help bust some trash! Below is our trash map. You can use it to mark areas on the map where you have spotted litter. Be sure to add a description if you drop a new pin! You can also delete pins to cleanup sites once all trash has been removed. Thank you for making our community more beautiful!
              </p>
            </div>
          </div>
        </div>
        <div
          class="box container"
          data-testid="trashmap-container"
          id="trashmap-container"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <h3
                class="text-center"
              >
                Trash Map
              </h3>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col"
              id="trashmap-container-col"
            >
              <div />
              <div>
                Loading...
              </div>
              <div
                class="spinner-border text-dark"
                data-testid="spinner"
                id="spinner"
              />
            </div>
          </div>
        </div>
        <div
          class="box container"
          data-testid="contact-container"
          id="contact-container"
        >
          <div
            class="row"
          >
            <h3
              class="text-center"
            >
              Contact Us
            </h3>
          </div>
          <div
            class="row"
          >
            <div
              class="d-flex justify-content-center col"
            >
              <form
                action="https://formspree.io/f/xyyaglza"
                class=""
                data-testid="form"
                id="form"
                method="POST"
              >
                <div
                  class="mb-3"
                >
                  <label
                    class="mb-2 form-label"
                    for="formBasicEmail"
                  >
                    Email address
                  </label>
                  <input
                    class="form-control"
                    id="formBasicEmail"
                    name="email"
                    placeholder="Enter email"
                    required=""
                    type="email"
                  />
                </div>
                <div
                  class="mb-3"
                >
                  <label
                    class="mb-2 form-label"
                    for="formBasicName"
                  >
                    Name
                  </label>
                  <input
                    class="form-control"
                    data-testid="name-input"
                    id="formBasicName"
                    name="name"
                    placeholder="Enter your name"
                    required=""
                    type="text"
                  />
                </div>
                <div
                  class="mb-3"
                >
                  <label
                    class="mb-2 form-label"
                    for="formBasicMessage"
                  >
                    Message
                  </label>
                  <input
                    class="form-control"
                    data-testid="message-input"
                    id="formBasicMessage"
                    name="message"
                    placeholder="Enter your message"
                    required=""
                    rows="3"
                    type="textarea"
                  />
                </div>
                <button
                  class="btn-lg btn btn-primary"
                  type="submit"
                >
                  Submit
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
